src/JoiForm.js

```diff
@@ -44,23 +44,9 @@ class JoiForm extends Component {
 
     const { schema = [], values } = props;
 
-    state.values = values
     state.schema = schema
     state.errors = {}
-
-    state.schema.forEach((fieldSchema) => {
-      const meta = merge(fieldSchema._meta);
-      const name = meta.name || camelize(fieldSchema._flags.label);
-
-      // if no value set for this field, but their is a default, set it
-      const hasDefault = state.values[name] === undefined && fieldSchema._flags.default !== undefined;
-      if (hasDefault) state.values[name] = fieldSchema._flags.default;
-
-      // if no value set for this field, but is boolean, set it to false
-      const setBoolean = state.values[name] === undefined && fieldSchema._type === 'boolean';
-      if (setBoolean) state.values[name] = false;
-    });
-
+    state.values = { ...defaultValues(schema, values), ...values}
```

Comment:
We should treat errors the same as values. If passed in pros they should override `state.errors`.